### ISSUE TITLE
Pasting via middle-click clears X selection

### ISSUE THREAD

**Issue by Markus-N** [geany/geany#3658]
I'm using geany 2.0-1 on Arch Linux with XFCE.

Since the newest update, I can not use paste via middle-click repeatedly anymore. This was a frequent use case for me: select some text and then paste it at a few places. It does not matter if the selection was made inside or outside geany.
What makes me think  geany clears the X selection:
* I can select and paste into a different window multiple times, but after pasting into geany, I can't anymore.
* Also, the clipboard manager plugin for the XFCE panel shows no mouse selection anymore afterwards.

I don't think this is a duplicate of https://github.com/geany/geany/issues/2629

**Comment by elextr** [comment]
Are you using Wayland or X11?

**Comment by b4n** [comment]
I don't see the same behavior, although kinda similar: I cannot paste multiple times from the editor to itself.  On the first paste it clears the initial selection, and that clears the PRIMARY X selection.  I believe that used to work, which was a lot more convenient I think.
However, with other apps (or documents) as either source or target I don't have the issue and can paste as many times as I want.

**Comment by elextr** [comment]
Yes, middle click pasting multiple in Geany used to work with 1.38, but I can test only on X11, not Wayland/Xwayland.  The change is deliberate in Scintilla (wade through the Scintilla bugs/fixes linked from #2629 to get https://sourceforge.net/p/scintilla/code/ci/921cb3dea3112a) and related to fixing something for Wayland selections.

**Comment by elextr** [comment]
I thought I saw a contribution by @eht16 so he may be able to add more info.

**Comment by Markus-N** [comment]
I'm using X11.  
AFAIK, XFCE does not yet support Wayland.

**Comment by eht16** [comment]
I can confirm the behavior @b4n described and this is the result of https://sourceforge.net/p/scintilla/bugs/2287/ which is the result of #2629.

The previous behavior (pasting the same selection multiple times) was more convenient, I agree. As Neil stated on https://sourceforge.net/p/scintilla/bugs/2287/#9b51, this would require more changes in Scintilla.

**Comment by eht16** [comment]
So in the end, it should be reported to Scintilla, I think.

**Comment by Markus-N** [comment]
I'd like to add that it's not only about convenience in geany. It also has an impact on the outside as the selection is lost completely and no more available for other applications too.

**Comment by cousteaulecommandant** [comment]
Can this be fixed (at least temporarily) the Gedit way?  I.e., when middle-click pasting, do NOT deselect the original text.

PS: As a curiosity, if you use the "Split window" plugin, selecting text on one panel and middle-clicking on the other will not clear the buffer (in either direction) so you can paste repeatedly, but selecting on one panel and middle-clicking on the same panel will clear the buffer (in either panel); at least on my Windows version.  (Not that this discovery helps a lot to fix the bug, but at least it's a workaround in case you want to paste multiple times and really want to use middle-click.)

**Comment by reiveune** [comment]
I am affected by this problem too, do you know if there is a fix planned in Scintilla in the future?

**Comment by elextr** [comment]
> I am affected by this problem too, do you know if there is a fix planned in Scintilla in the future?

You would have to ask Scintilla, it is a separate project not part of Geany.

**Comment by reiveune** [comment]
It's great to see that the problem is fixed in Scintilla, so we just have to wait for the next release of Geany that includes updated Scintilla. Anyone have any idea when the next release will be?
